handler/src/main/java/io/netty/handler/ssl/SslHandlerCoalescingBufferQueue.java

```diff
@@ -76,6 +79,11 @@ protected ByteBuf composeFirst(ByteBufAllocator allocator, ByteBuf first) {
         return first;
     }
 
+    @Override
+    protected ByteBuf returnFirst(ByteBufAllocator allocator, ByteBuf first) {
+        return SslHandler.cloneBufferIfReadOnlyAndWantsMutable(allocator, first, wantsMutableBuffer, wantsDirectBuffer);
```

Comment:
I think should better move the `cloneBufferIfReadOnlyAndWantsMutable` in this class so we don't have a 'cycle-dependency' between this class and `SslHandler`

Reply:
@normanmaurer Makes sense.
However, I'm planning to close this PR as something that isn't needed. I'll add a separate comment while closing.